app/models/classification.rb

```diff
@@ -3,27 +3,29 @@ class Classification < ApplicationRecord
 
   belongs_to :tag
 
+  virtual_column :name, :type => :string
+
   before_save    :save_tag
   before_destroy :delete_tags_and_entries
-  validate       :validate_format_of_name, :validate_uniqueness_on_tag_name
-
-  validates_uniqueness_of :description, :scope => [:parent_id],
-                                        :if    => proc { |c|
-                                          cond = ["parent_id = ? AND description = ?", c.parent_id, c.description]
-                                          unless c.new_record?
-                                            cond.first << " AND id != ?"
-                                            cond << c.id
-                                          end
-                                          c.class.in_region(region_id).exists?(cond)
-                                        }
-  validates_presence_of :name, :description
-  validates_length_of :name, :maximum => 50, :message => "must not exceed 50 characters"
-  validates_length_of :description, :maximum => 255, :message => "must not exceed 255 characters"
-  validates_inclusion_of :syntax,
-                         :in      => %w( string integer boolean ),
-                         :message => "should be one of 'string', 'integer' or 'boolean'"
 
-  virtual_column :name, :type => :string
+  validates :description, :presence => true, :length => {:maximum => 255}
+  validates :description, :uniqueness => {:scope => [:parent_id]}, :if => proc { |c|
+    cond = ["parent_id = ? AND description = ?", c.parent_id, c.description]
+    unless c.new_record?
+      cond.first << " AND id != ?"
+      cond << c.id
+    end
+    c.class.in_region(region_id).exists?(cond)
+  }
```

Comment:
[I just rewrote these to `validates :foo, :length` style because Rubocop was angry at `validates_length_of` style.
My only intentional change here is NAME_MAX_LENGTH const for new `sanitize_name()` method.]

BTW while you're looking at this validation, it has a bug/misfeature (irrelevant to this PR):
It doesn't allow 2 categories with same description even under different tag namespaces (/managed/cat vs /namespace2/cat).
The reason is that classifications table is unaware of namespaces.  A category's parent is not the namespace — there are no explicit tags nor classifications rows for `/managed` or `/other` — their parent_id is 0, so all categories are in one scope here.
The only way to know a classification's namespace is chunking the associated `.tag.name` on slashes, which is what `.ns` ends up doing...